### ISSUE TITLE
False positives for RedundantCondition and UnusedVariable when using globals

### ISSUE THREAD

**Issue by kkmuffme** [vimeo/psalm#2607]
It seems psalm doesn't pick up the use of the global variable:
(this code is a simplified version of what the real code is to easily visualize the problem)

```
ERROR: RedundantCondition - 14:8 - Found a redundant condition when evaluating $contains_int and trying to reconcile type 'false' to false

INFO: UnusedVariable - 32:5 - Variable $contains_int is never referenced
```

https://psalm.dev/r/ca52559db3

```php
<?php

$arg = array( 'hello', 'world' );
start_here( $arg );

/**
 * @param array $arg
 * @return void
 */
function start_here( $arg ) {
  global $contains_int;
  $contains_int = false;
  array_walk_recursive( $arg, 'check_if_array_contains_int' );
  if ( $contains_int === false ) {
    echo "hello";
  }
}

/**
 * callback for array_walk_recursive to check if array contains any objects
 *
 * @param mixed $value the array value - we will only modify strings though
 * @return void
 */
function check_if_array_contains_int( $value ) {
  global $contains_int;
  if ( $contains_int === true ) {
    return;
  }

  if ( is_int( $value ) ) {
    $contains_int = true;
  } 
}
```

**Comment by muglug** [comment]
Just suppress the issue – I'm not going to spend time on this, sorry (using `global` like this, while valid, is now considered incredibly archaic and a bad way of writing PHP)
